server/src/main/java/org/opensearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -264,7 +265,7 @@ private class PendingMerges implements Releasable {
         private final SearchPhaseController.TopDocsStats topDocsStats;
         private volatile MergeResult mergeResult;
         private volatile boolean hasPartialReduce;
-        private volatile int numReducePhases;
+        private final AtomicInteger numReducePhases = new AtomicInteger();
```

Comment:
Do we still need to change this to AtomicInteger?

If usage of an int hasn't been resulted in any issue so far as we know, then I think we might just add unnecessary overhead of using AtomicInteger over an int.

Reply:
There is a definite issue here. But no one digs that deeply and reports it, if faces.
++num and num++ operations aren't atomic, so you can get an out-of-sync value from time to time

Reply:
I don't think your analysis is entirely accurate. 

The increment operation (++numReducePhases) is performed within a thread-safe block. Specifically, `tryExecuteNext()` method begins by acquiring a synchronized lock on `this`.

We’d benefit from `AtomicInteger` instead of `volatile int` if `numReducePhases` were incremented outside of a synchronized block or lock-free code.

Reply:
@sandeshkr419 You are right. That was the only previously approved change and it's uneecessary. I'm going to close this PR